dali/operators/reader/loader/video_loader.cc

```diff
@@ -639,6 +639,8 @@ void VideoLoader::read_file() {
       continue;
     }
 
+    DALI_ENFORCE(pkt, "Failed to read frames.");
+
```

Comment:
If the `if` condition from line 585 is not true, the `pkt` pointer is left initialised to `nullptr`.

Reply:
In that case L609 should kick in and we end up either in L615 or L639.

Reply:
If there are no case where neither of L585 nor L606 ifs are entered, then maybe the second if should just be an `else` to the first one?

Reply:
We start with `int last_key_frame = -1;`, the very first read in while loop fails so that `read_frame_ret < 0`. The `last_key_frame` remains `-1` and 609 is `false && true`, it seems that the ptr really remains to be nullptr.

Reply:
> We start with `int last_key_frame = -1;`, the very first read in while loop fails so that `read_frame_ret < 0`. The `last_key_frame` remains `-1` and 609 is `false && true`, it seems that the ptr really remains to be nullptr.

 I agree. I have been thinking that maybe we can initialize `dec_status` with `REQ_NOT_STARTED` instead, but I'm not 100% confident there will be no side effect. Let it be as @banasraf proposed.